src/script/arch-update-tray.py

```diff
@@ -25,22 +25,24 @@
 
 def arch_update():
     """ Launch with desktop file """
+    DESKTOP_FILE = None
     if 'XDG_DATA_HOME' in os.environ:
         DESKTOP_FILE = os.path.join(
             os.environ['XDG_DATA_HOME'], 'applications', 'arch-update.desktop')
-    if not os.path.isfile(DESKTOP_FILE):
+    if not DESKTOP_FILE or not os.path.isfile(DESKTOP_FILE):
         if 'HOME' in os.environ:
             DESKTOP_FILE = os.path.join(
                 os.environ['HOME'], '.local', 'share', 'applications', 'arch-update.desktop')
-    if not os.path.isfile(DESKTOP_FILE):
+    if not DESKTOP_FILE or not os.path.isfile(DESKTOP_FILE):
         if 'XDG_DATA_DIRS' in os.environ:
             DESKTOP_FILE = os.path.join(
                 os.environ['XDG_DATA_DIRS'], 'applications', 'arch-update.desktop')
-    if not os.path.isfile(DESKTOP_FILE):
+    if not DESKTOP_FILE or not os.path.isfile(DESKTOP_FILE):
         DESKTOP_FILE = "/usr/local/share/applications/arch-update.desktop"
     if not os.path.isfile(DESKTOP_FILE):
         DESKTOP_FILE = "/usr/share/applications/arch-update.desktop"
-    subprocess.run(["gio", "launch", DESKTOP_FILE], check=False)
+    if os.path.isfile(DESKTOP_FILE):
```

Comment:
Whit this check, won't it fail silently if the file isn't found? If so, my opinion is to get rid of this check/condition so if the file isn't found, at least we have `gio` throwing an error about it.

Reply:
Or maybe add an `else` statement to handle such a case?

Reply:
The idea was to have a way to show some logs in case there's ever an opened issue like "I click on the systray but nothing happens", so we could ask them to launch the systray from the terminal and get such an error message. As it is very unlikely that a missing desktop file would ever happen, I personally feel like a console error is enough we can point people to is enough.

But if a graphical error is easy to do and do not require additional dependencies and stuff, why not ^^

Reply:
It would likely need us to bring in the translation code again.

> I personally feel like a console error we can point people to is enough
 
Fair enough, we'll let gio throw the error for us then